multi-agent/lib/mas/blueprints/service.py

```
@@ -143,44 +170,77 @@ def list_draft_docs(
 
         for doc in docs:
             try:
-                resolved_docs.append(self._resolve_doc(doc))
+                resolved_docs.append(self._resolve_doc(doc, caller=caller))
             except Exception as e:
-                print(f"Skipping blueprint '{doc.blueprint_id}': resolution failed — {e}")
+                logger.warning(
+                    "Skipping blueprint '%s': resolution failed — %s",
+                    doc.blueprint_id, e,
+                )
                 continue
```

Ruff (0.16.1)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)